azure/services/natgateways/client.go

```diff
@@ -33,26 +33,17 @@ type azureClient struct {
 	natgateways *armnetwork.NatGatewaysClient
 }
 
-// newClient creates a new VM client from subscription ID.
+// newClient creates a new nat gateways client from an authorizer.
```

Comment:
Are these changes supposed to be a part of this PR? Or did it get mixed in from #3858?

Reply:
This is a secondary commit that fixes some typos and an incorrect comment that I noticed after #3858 merged. It also does minor simplification of the `newClient` func to make it consistent with this PR's approach.

I can make a separate PR from that if that's better, lmk.

Reply:
I think I will make a separate PR from that commit, actually. I'd like to make a similar simplification to the virtualmachineimages client, and I want this one to be fairly clean so we can point at it as an example of how to convert services to SDK v2 + `asyncpoller`.